### ISSUE TITLE
Cannot assign DHCP addresses to In-band ports

### ISSUE THREAD

**Issue by ciju-juniper** [Azure/sonic-buildimage#5732]
**Problem description**
In the recent SONiC images, DHCP address assignment for in-band ports doesn't work. This is a blocker for in-band ztp. Once the ZTP kicks in, it tries to get a DHCP address over the management / in-band ports. DHCP does work over management port but it fails to assign an address to in-band ports.

From the switch I can see it requests for a DHCP address:
 
Oct  7 11:18:43.863759 sonic INFO dhclient[4393]: DHCPDISCOVER on Ethernet88 to 255.255.255.255 port 67 interval 19
Oct  7 11:19:02.868723 sonic INFO dhclient[4393]: No DHCPOFFERS received.
Oct  7 11:19:02.869300 sonic INFO dhclient[4393]: No working leases in persistennt database - sleeping.
 
In the server, it gets the request and offers an IP
 
Oct 07 22:33:44 fever dhcpd[177619]: DHCPDISCOVER from 0c:59:9c:38:ff:cb via enp216s0f0
Oct 07 22:33:44 fever dhcpd[177619]: DHCPOFFER on 2.2.2.10 to 0c:59:9c:38:ff:cb via enp216s0f0

The OFFER doesn't reach the upper layers in the switch and it gets dropped at the ASIC.

**Testing environment**
Juniper QFX5210 / QFX5200 platforms built from master & 201911 branches, directly connected to a Ubuntu-16.04 server.

**Root cause for this issue**
There was a commit of copp.json file for control plane policing through the PR https://github.com/Azure/sonic-buildimage/pull/5053 which specifically enabled DHCP only for 'ToRRouter' type of switches.  This condition check will drop the DHCP packets for other types of switches (Eg: 'LeafRouter')

`{% if DEVICE_METADATA is defined and DEVICE_METADATA['localhost'] is defined and DEVICE_METADATA['localhost']['type'] is defined and DEVICE_METADATA['localhost']['type'] != "ToRRouter" %}`

**Reasons why should it be fixed**

1. ZTP should be functional for all types of switches such as TOR, LEAF, etc. Why do we want to restrict ZTP only for TOR switches?
2. AFAIK, the 'DEVICE_METADATA' block is generated automatically at the build time, and there is no way that a platform can have a setting in the platform directory to specify the contents that should go to 'DEVICE_METADATA'.
3. Another point is, nobody want to build different SONiC images for TOR, LEAF switches by specifying the type of switch at the build time.

**Comment by ciju-juniper** [comment]
@lguohan @abdosi @rajendra-dendukuri  Please have your thoughts on this.

**Comment by lguohan** [comment]
i think you are right, but a more complete solution is to disable dhcp copp after ztp is finished. This is because dhcp packets should not be trapped on leaf router. currently copp rule is not configurable. @prsunny , will coppmgr allow copp rule to be configurable. also, @rajendra-dendukuri , is ztp supported over in-band interface?

**Comment by ciju-juniper** [comment]
@lguohan in-band ztp is supported and I'm using it successfully now: https://github.com/Azure/SONiC/blob/master/doc/ztp/ztp.md#37-provisioning-over-in-band-network

**Comment by ciju-juniper** [comment]
@prsunny @rajendra-dendukuri  Please take a look at this diff and review the changes.

I will create a pull request to sonic-buildimage once after current PR is merged. I haven't had branches created before creating the initial pull request.

**Comment by ciju-juniper** [comment]
@prsunny Could you take a look at the proposed suggestion? If you are fine with it, I will raise a PR to sonic-buildimage repo.

**Comment by prsunny** [comment]
With the new coppmgr changes, the traps can be installed or removed (via config_db), does that help solve this? Because as Guohan mentioned, we don't want to keep it installed in case of LeafRouter but remove the trap once ZTP is done. @dgsudharsan

**Comment by ciju-juniper** [comment]
@prsunny When the ztp is enabled at build time, there will not be any config_db.json at the first boot. What will happen in that case? Wouldn't the DHCP be disabled?

**Comment by ciju-juniper** [comment]
@prsunny Could the DHCP be enabled by default if there is no config_db.json present? And the new config_db deciding whether to enable DHCP or not ?

**Comment by judyjoseph** [comment]
Please create a new issue after trying to enable dhcp-relay feature and verify.

**Comment by prsunny** [comment]
Closing this since its old and no further action on this.
